src/clang-cpp-frontend/clang_cpp_convert.cpp

```diff
@@ -976,6 +976,34 @@ bool clang_cpp_convertert::get_expr(const clang::Stmt &stmt, exprt &new_expr)
     break;
   }
 
+  case clang::Stmt::LambdaExprClass:
+  {
+    const clang::LambdaExpr &lambda_expr =
+      static_cast<const clang::LambdaExpr &>(stmt);
+
+    if (lambda_expr.capture_size() > 0)
+    {
+      log_error("ESBMC does not support lambdas with captures for now");
+      return true;
+    }
+
+    get_struct_union_class(*lambda_expr.getLambdaClass());
+
+    // Construct a new object of the lambda class
+    typet lambda_class_type;
+    if (get_type(
+          *lambda_expr.getLambdaClass()->getTypeForDecl(), lambda_class_type))
+      return true;
+    new_expr = gen_zero(pointer_typet(lambda_class_type));
```

Comment:
It might be easy to extend it later if we do this:
```cpp
if (lambda_expr.capture_size())
{
    log_error("ESBMC does not support lambdas with captures for now");
    return true;
}
else
     new_expr = gen_zero(pointer_typet(lambda_class_type));
```